magnum_cluster_api/resources.py

```diff
@@ -891,6 +915,238 @@ def get_object(self) -> objects.ClusterClass:
+                                                        systemd:
+                                                          units:
+                                                          - name: coreos-metadata.service
+                                                            dropins:
+                                                            - name: 20-clct-provider-override.conf
+                                                              contents: |
+                                                                [Service]
+                                                                # Set Openstack as coreos-metadata provider
+                                                                Environment=COREOS_METADATA_OPT_PROVIDER=--provider=openstack-metadata
+                                                          - name: coreos-metadata-sshkeys@.service
+                                                            enabled: true
+                                                            dropins:
+                                                            - name: 20-clct-provider-override.conf
+                                                              contents: |
+                                                                [Service]
+                                                                # Set Openstack as coreos-metadata provider
+                                                                Environment=COREOS_METADATA_OPT_PROVIDER=--provider=openstack-metadata
+                                                          - name: kubeadm.service
+                                                            dropins:
+                                                            - name: 10-flatcar.conf
+                                                              contents: |
+                                                                [Unit]
+                                                                # kubeadm must run after coreos-metadata populated /run/metadata directory.
+                                                                Requires=coreos-metadata.service
+                                                                After=coreos-metadata.service
+                                                                [Service]
+                                                                # Ensure kubeadm service has access to kubeadm binary in /opt/bin on Flatcar.
+                                                                Environment=PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/bin
+                                                                # To make metadata environment variables available for pre-kubeadm commands.
+                                                                EnvironmentFile=/run/metadata/*
+                                                        """  # noqa: E501
```

Comment:
this feels like it should live inside `image-builder`, so I think as a separate PR we should fix this?

Reply:
The PATH modifying part should not be needed anymore since https://github.com/kubernetes-sigs/image-builder/commit/f4864de5cad4a826aff80259bf6681e37a49f18d, but I think dependence on `coreos-metadata.service` is platform-specific, so I don't think `image-builder` is a good place to put this.